kale/embed/mpca.py

```diff
@@ -1,49 +1,76 @@
 # =============================================================================
 # Author: Shuo Zhou, szhou20@sheffield.ac.uk
 #         Haiping Lu, h.lu@sheffield.ac.uk
-#
-# Python implement of Multilinear Principal Component Analysis (MPCA)
-#
-# Ref: Haiping Lu, K.N. Plataniotis, and A.N. Venetsanopoulos, "MPCA: Multilinear
-# Principal Component Analysis of Tensor Objects", IEEE Transactions on Neural 
-# Networks, Vol. 19, No. 1, Page: 18-39, January 2008.
 # =============================================================================
+
+"""Python implementation of Multilinear Principal Component Analysis (MPCA)
+Includeing implementaion as a scikit-learn object and an independent function.
+
+Reference:
+    Haiping Lu, K.N. Plataniotis, and A.N. Venetsanopoulos, "MPCA: Multilinear
+    Principal Component Analysis of Tensor Objects", IEEE Transactions on Neural 
+    Networks, Vol. 19, No. 1, Page: 18-39, January 2008. For initial Matlab
+    implementation, please go to https://uk.mathworks.com/matlabcentral/fileexchange/26168.
+"""
+
 import numpy as np
-import tensorly as tl
-from tensorly.base import fold, unfold
-from tensorly.tenalg import mode_dot, multi_mode_dot
+# import tensorly as tl
+from tensorly.base import unfold
+from tensorly.tenalg import multi_mode_dot
 from sklearn.base import BaseEstimator, TransformerMixin
 
 
 class MPCA(BaseEstimator, TransformerMixin):
-    def __init__(self, variance_explained=0.97, max_iter=1):
-        """
-        Implementation compatible with sickit-learn
-        ----------
-        Parameter:
-            variance_explained: ration of variance to keep (between 0 and 1)
-            max_iter: max number of iteration, integer
-        ----------
-        Attributes:
-            tPs: list of projection matrices
-        """
-        self.var_exp = variance_explained
+    """MPCA implementation compatible with sickit-learn
+
+    Args:
+        var_explained (float, optional): ration of variance to keep
+            (between 0 and 1). Defaults to 0.97.
+        max_iter (int, optional): max number of iteration. Defaults to 1.
+
+    Attributes:
+        tPs: list of projection matrices
```

Comment:
Yes, I used it in the Matlab code but now it is time to make it **more readable** like those in sklearn https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html . 't' stands for transposed (see matlab code _tUs{n}=Us{n}'_. Therefore, we can call it _transposed_proj_mats_ or _trans_proj_mats_ or _transp_proj_mats_

The doc can also be improved. In matlab 
tUs: the multilinear projection, consisting of N projection matrices, one for each mode
Will it be better to mention the same that there are N of them, one for each mode?

Reply:
Will change later.

Reply:
Maybe just *proj_mats*? We can explain it is transposed in docstrings.